modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,728 @@
+	$initstr .= '<!-- generator="jetpack-' . JETPACK__VERSION . '" -->' . "\n";
+	$initstr .= '<urlset';
+	foreach ( jetpack_sitemap_namespaces() as $attribute => $value ) {
+		$initstr .= ' ' . esc_html( $attribute ) . '="' . esc_attr( $value ) . '"';
+	}
+	$initstr .= ' />';
+
+	return $initstr;
+}
+
+/**
+ * Load XSLT for sitemap.
+ *
+ * @module xml-sitemap
+ *
+ * @param string $type XSLT to load.
+ */
+function jetpack_load_xsl( $type = '' ) {
+
+	$transient_xsl = empty( $type ) ? 'jetpack_sitemap_xsl' : "jetpack_{$type}_sitemap_xsl";
+
+	$xsl = get_transient( $transient_xsl );
+
+	if ( $xsl ) {
+		header( 'Content-Type: ' . jetpack_sitemap_content_type(), true );
+		echo $xsl;
+		die();
+	}
+
+	// Populate $xsl. Use $type.
+	include_once JETPACK__PLUGIN_DIR . 'modules/xml-sitemap/sitemap-xsl.php';
+
+	if ( ! empty( $xsl ) ) {
+		set_transient( $transient_xsl, $xsl, DAY_IN_SECONDS );
+		echo $xsl;
+	}
+
+	die();
+}
+
+/**
+ * Responds with an XSLT to stylize sitemap.
+ *
+ * @module xml-sitemap
+ */
+function jetpack_print_sitemap_xsl() {
+	if ( defined( 'JETPACK_SKIP_DEFAULT_SITEMAP' ) && JETPACK_SKIP_DEFAULT_SITEMAP ) {
+		return;
+	}
+	jetpack_load_xsl();
+}
+
+/**
+ * Responds with an XSLT to stylize news sitemap.
+ *
+ * @module xml-sitemap
+ */
+function jetpack_print_news_sitemap_xsl() {
+	if ( defined( 'JETPACK_SKIP_DEFAULT_NEWS_SITEMAP' ) && JETPACK_SKIP_DEFAULT_NEWS_SITEMAP ) {
+		return;
+	}
+	jetpack_load_xsl( 'news' );
+}
+
+/**
+ * Print an XML sitemap conforming to the Sitemaps.org protocol.
+ * Outputs an XML list of up to the latest 1000 posts.
+ *
+ * @module xml-sitemap
+ *
+ * @link http://sitemaps.org/protocol.php Sitemaps.org protocol.
+ */
+function jetpack_print_sitemap() {
+	if ( defined( 'JETPACK_SKIP_DEFAULT_SITEMAP' ) && JETPACK_SKIP_DEFAULT_SITEMAP ) {
+		return;
+	}
+	global $wpdb;
+
+	$xml = get_transient( 'jetpack_sitemap' );
+
+	if ( $xml ) {
+		header( 'Content-Type: ' . jetpack_sitemap_content_type(), true );
+		echo $xml;
+		die();
+	}
+
+	/**
+	 * Filter the post types that will be included in sitemap.
+	 *
+	 * @module xml-sitemap
+	 *
+	 * @since 3.9
+	 *
+	 * @param array $post_types Array of post types.
+	 */
+	$post_types    = apply_filters( 'jetpack_sitemap_post_types', array ( 'post', 'page' ) );
+
+	$post_types_in = array ();
+	foreach ( (array) $post_types as $post_type ) {
+		$post_types_in[] = $wpdb->prepare( '%s', $post_type );
+	}
+	$post_types_in = join( ",", $post_types_in );
+
+	// use direct query instead because get_posts was acting too heavy for our needs
+	//$posts = get_posts( array( 'numberposts'=>1000, 'post_type'=>$post_types, 'post_status'=>'published' ) );
+	$posts = $wpdb->get_results( "SELECT ID, post_type, post_modified_gmt, comment_count FROM $wpdb->posts WHERE post_status='publish' AND post_type IN ({$post_types_in}) ORDER BY post_modified_gmt DESC LIMIT 1000" );
+	if ( empty( $posts ) ) {
+		header( 'HTTP/1.0 404 Not Found', true, 404 );
```

Comment:
We could use `status_header()` instead for those occasions someone is filtering them for whatever reason.